index.js

```diff
@@ -2,74 +2,108 @@
 var isRegexp = require('is-regexp');
 var isPlainObj = require('is-plain-obj');
 
-module.exports = function (val, opts, pad) {
-	var seen = [];
-
-	return (function stringify(val, opts, pad) {
-		opts = opts || {};
-		opts.indent = opts.indent || '\t';
-		pad = pad || '';
-
-		if (val === null ||
-			val === undefined ||
-			typeof val === 'number' ||
-			typeof val === 'boolean' ||
-			typeof val === 'function' ||
-			isRegexp(val)) {
-			return String(val);
-		}
+var stringify = module.exports = function (val, opts, pad, seen) {
+	opts = opts || {};
+	opts.indent = opts.indent || '\t';
+	pad = pad || '';
+	seen = seen || [];
+
+	var type = stringify.type(val);
+	return stringify[type](val, opts, pad, seen);
+};
+
+stringify.typeOrder = [
```

Comment:
We now define types in 3 places. Here, `stringify.types` and the `exports`. Would be nice if we could bring all these together into one somehow.